examples/esp-todomvc-react/js/model/eventProcessors/TodoListEventProcessor.js

```diff
@@ -0,0 +1,158 @@
+/*global esp, app*/
+
+(function () {
+    "use strict";
+
+    var extend = app.Utils.extend;
+
+    var TodoListEventProcessor = function (router) {
+        esp.model.DisposableBase.call(this);
+        this.router = router;
+    };
+
+    TodoListEventProcessor.prototype = Object.create(esp.model.DisposableBase.prototype);
+
+    TodoListEventProcessor.prototype.start = function () {
+        this.observeInitEvent();
+        this.observeToggleEvent();
+        this.observeToggleAllEvent();
+        this.observeTodoDestroyedEvent();
+        this.observeTodoAddedEvent();
+        this.observeFilterChangedEvent();
+        this.observeClearCompletedEvent();
+        this.observeEditStartedEvent();
+        this.observeEditCancelledEvent();
+        this.observeEditCompletedEvent();
+    };
+
+    TodoListEventProcessor.prototype.observeInitEvent = function () {
+        this.addDisposable(this.router
+            .getEventObservable("todoList", "initEvent")
+            .observe(function (model) {
+                var todoItems = app.Utils.store(model.localStorageKey);
+                var todoItemsById = {};
+                if (todoItems) {
+                    todoItems.forEach(function (todoItem) {
+                        todoItemsById[todoItem.id] = todoItem;
+                    });
+                    model.todoItemsById = todoItemsById;
+                }
+            })
+            );
+    };
+
+    TodoListEventProcessor.prototype.observeToggleEvent = function () {
+        this.addDisposable(this.router
+            .getEventObservable("todoList", "todoToggled")
+            .observe(function (model, event) {
+                var todoItem = model.todoItemsById[event.id];
+                todoItem = extend({}, todoItem, { complete: !todoItem.complete });
+                model.todoItemsById[todoItem.id] = todoItem;
```

Comment:
Not sure if esp-js does this, but we should have a clone of the model we manipulate or use the React immutability helpers. The save method clones then the model is returned but the same model which is currently rendered is being mutated at the moment?